packages/react-menu/src/components/MenuTrigger/useTriggerElement.ts

```diff
@@ -135,7 +135,7 @@ export const useTriggerElement = (state: MenuTriggerState): MenuTriggerState =>
 
   state.children = React.cloneElement(child, {
     ...triggerProps,
-    ref: useMergedRefs(child.props.ref, triggerRef),
+    ref: useMergedRefs(((child as unknown) as { ref: React.Ref<unknown> }).ref, triggerRef),
```

Comment:
Oops, yeah, `.ref` is not a prop 😅

---

- Ideally we should use there `React.ReactElement` the problem is that it does not contain `ref` 😯
- nit: I suggest to use `React.ReactElement` & `React.RefAttributes`

```suggestion
    ref: useMergedRefs(((child as unknown) as React.ReactElement & React.RefAttributes<unknown>).ref, triggerRef),
```

As `forwardRef()` in types uses similar definition:

```
    function forwardRef<T, P = {}>(render: ForwardRefRenderFunction<T, P>): ForwardRefExoticComponent<PropsWithoutRef<P> & RefAttributes<T>>;
```